java-symbol-solver-core/src/main/java/me/tomassetti/symbolsolver/javaparsermodel/JavaParserFacade.java

```diff
@@ -531,6 +527,14 @@ private Type getTypeConcrete(Node node, boolean solveLambdas) {
         }
     }
 
+    public Type convertToUsageVariableType(VariableDeclarationExpr var) {
+        Type type = JavaParserFacade.get(typeSolver).convertToUsage(var.getElementType(), var);
+        for (int i=0;i<var.getArrayBracketPairsAfterElementType().size();i++) {
+            type = new ArrayType(type);
```

Comment:
This code is full of bugs :) VariableDeclarationExpr can contain multiple variables with varying amounts of brackets. There is not one type. You can iterate the variables in it, and getType() on them - you will get the correct types.